### ISSUE TITLE
Launcher closes directly after launching on a new PC?

### ISSUE THREAD

**Issue by VIERcntHOLZ** [Healix/Gw2Launcher#281]
Hey @Healix,

yesterday I upgraded my PC, fresh new windows 10 install - restored a backup I created yesterday with Gw2LauncherBeta (where everything went super smooth) - now having my alts (using a distinct .dat for each alt) back in the Launcher. When I tried launching one, options popped up and he wanted me to select the new Gw2-64.exe, set the new Path, worked fine.

Now, when I start my alt, he pulls up the gw launcher and closes it directly again in like 0.1 Seconds, like instantly.
I went into the Options for one Account => Launch Options => unticked "Autmatically login when remembered" => Pulls up the launcher with correct login credentials in it - but, when I hit "login" => once again 0.1 Second but shows the blue box of "Guild wars has encountered an error" message with "The Account Name and Password youve entered is invalid".
Yesterday everything worked fine.

Do I really now need to re do all the (50) dats because the backup corrupted the login/password credentials?

**Comment by VIERcntHOLZ** [comment]
just tried to enter my password manually after i started it w/o the "Automatically Login when remembered" => this then launches the alt normally.
When I close this instance again and restart said instance, it once again fails the login and closes itself.

Seems like the .dat files arent having the passwords included anymore?

**Comment by Healix** [comment]
Did the logins never work on the new install? It sounds like it was working at first.

You can disable automatic logins by right clicking any account > All/Selected > Disable automatic logins.

If you have the login timeout enabled, it will automatically relaunch GW2 if the login fails. That should be the only reason why GW2 would instantly close.

Can you see that the remembered email is correct? It is possible for GW2 to corrupt the remembered password. If it's corrupted, you'll have to set it again. There is also a bug that GW2 has had since July that can cause the email and/or password to randomly not load or to partially load. It's fairly rare to encounter, but for some, it's a common problem.

The remembered login is handled by GW2 and stored in Local.dat. Remember, GW2 is in read-only mode when launching an account, so if you want to change it, you have to right click > Selected > Launch (normal), then you have to successfully login and close the game normally (can't force quit/kill it) for it to save.

**Comment by VIERcntHOLZ** [comment]
On my New PC/Fresh Windows install this error occured from the very beginning - yesterday (old machine) they worked perfectly fine.

The remembered E-Mails were correct - had to re-do all the 50 .dat files unfortunately, now, with re-set .dats it works as intended.
Seems like the backuped .dats all had a corrupted pw then?

**Comment by Healix** [comment]
I tested it - the password in Local.dat is now encrypted to your user, so it'll be corrupted anywhere else.

**Comment by VIERcntHOLZ** [comment]
To the Windowsuser?
If thats the case, the backup/restoration function is rendered a bit "useless", or? (And I mean that in the most positive way possible, just thinking of the most logical way why one would use a backup/restore function)

**Comment by Healix** [comment]
Yes, it's encrypted to your user account in Windows. Originally anyone with the Local.dat file could use it to login. It was likely changed around the Steam launch; I last tested it about a year or two ago.

If your only reason for a backup is the password in Local.dat, then yes, it'd be useless, but it's going to be faster to have it so you only have to enter the password. If you had Gw2Launcher store the login instead of GW2, then it would have transferred, plus you wouldn't need a Local.dat per account.

**Comment by VIERcntHOLZ** [comment]
Got it - thank you

But having the disctinct local.dat file per account seemed to be "smoother"/"faster" than having gw2launcher copy/paste the passwords for each and every account, or? 
At least thats what I red like half a year ago when I set it up first time

**Comment by Healix** [comment]
Having GW2 remember the login is the safe way. It's normally more reliable (it's currently very unreliable for some people), will be about a second faster to login and won't break when ArenaNet rarely changes something. The downside is Local.dat is 20~40 MB each and it has to be updated with every patch, but that at least is solved with the option to allow direct updates.

You could always use both. You can have Gw2Launcher store the login, but not have the automatic entry enabled, then if GW2 loses the password, you only need to enable it and use launch (normal) to have it overwrite it.

**Comment by VIERcntHOLZ** [comment]
Wow, the second paragraph is a game changer - never thought of it that way - thank you! :)

**Comment by VIERcntHOLZ** [comment]
hey @Healix,
I might switch over to Win11 - is GW2Launcher supported?
And if so, were you able to fix the "export Backup"/Import Backup on new OS problem with the encrypted usernames/passwords?
Ive set the settings in the launcher => Security => Encryption => Unencrypted, will that help after changing the OS?

If not, can you give me a quick glimpse how I should prepare my .dat files?
Simply add the credentials for each account in the settings of said account?

**Comment by Healix** [comment]
There shouldn't be any problems with 11.

For passwords stored by Gw2Launcher, changing the encryption to unencrypted or portable will allow them to be moved to a new OS. You can just create a backup and restore it. You could also export the accounts for an extra backup.

For passwords remembered by GW2, they're encrypted to your user, so assume they will be lost.

**Comment by VIERcntHOLZ** [comment]
You sir, are a wizard!
Thanks a bunch!
